File: src/models/tweet.rb
require 'dotenv/load'
require_relative '../config/sequel'

class Tweet < Sequel::Model
  TOPICS = %w[nasa healthcare opensource]

  dataset_module do
    def latest_by_topic(topic)
      where(topic: topic).order(Sequel.desc(:id)).limit(10)
    end
  end
end

File: src/app/models/tweet.ts
import { Base } from "./base"
import { Expose } from "class-transformer"

export enum TweetTopic {
  Nasa = "nasa",
  Healthcare = "healthcare",
  Opensource = "opensource"
}

export class Tweet extends Base  {

  @Expose()
  public text: string

  @Expose()
  public topic: TweetTopic

  @Expose()
  public username: string

  @Expose({ name: "external_id" })
  public externalId: Number

}
